### ISSUE TITLE
'Invalid key set format' error when logging in

### ISSUE THREAD

**Issue by robintw** [djpugh/fastapi_aad_auth#87]
I have previously deployed fastapi_aad_auth successfully in an Azure Functions app. I'm now trying to replicate this in a second Azure Functions app, and am running into an error.

When I go to the `/login` endpoint, I get the standard dialogs to login, but then I get a 500 error. The full trace is below (sorry for the poor formatting, Azure provides the logs from Azure Functions in a strange format and I had to try and tidy it up myself) - the main error is an `Invalid key set format` error.

Do you have any idea what's going on here? I'm pretty sure I've set up the app in the same way as my first app (which works fine).

```
Exception: ValueError: Invalid key set format
Stack:

File "/azure-functions-host/workers/python/3.8/LINUX/X64/azure_functions_worker/dispatcher.py", line 402, in _handle__invocation_request
call_result = await self._loop.run_in_executor(
    
File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
        result = self.fn(*self.args, **self.kwargs)
    
File "/azure-functions-host/workers/python/3.8/LINUX/X64/azure_functions_worker/dispatcher.py", line 611, in _run_sync_func
        return ExtensionManager.get_sync_invocation_wrapper(context,
File "/azure-functions-host/workers/python/3.8/LINUX/X64/azure_functions_worker/extension.py", line 215, in _raw_invocation_wrapperresult = function(**args)
File "/home/site/wwwroot/HttpTrigger1/__init__.py", line 172, in mainreturn AsgiMiddleware(app).handle(req, context)
File "/home/site/wwwroot/HttpTrigger1/__init__.py", line 154, in handlereturn self._handle(req, context)
File "/home/site/wwwroot/HttpTrigger1/__init__.py", line 161, in _handleasgi_response = self._loop.run_until_complete(
File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_completereturn future.result()
File "/home/site/wwwroot/HttpTrigger1/__init__.py", line 71, in from_appawait app(scope, res._receive, res._send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi/applications.py", line 208, in __call__await super().__call__(scope, receive, send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/applications.py", line 112, in __call__await self.middleware_stack(scope, receive, send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/middleware/errors.py", line 181, in __call__raise exc
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/middleware/errors.py", line 159, in __call__await self.app(scope, receive, _send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/middleware/sessions.py", line 77, in __call__await self.app(scope, receive, send_wrapper)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/middleware/authentication.py", line 48, in __call__await self.app(scope, receive, send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/middleware/cors.py", line 84, in __call__await self.app(scope, receive, send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/exceptions.py", line 82, in __call__raise exc
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/exceptions.py", line 71, in __call__await self.app(scope, receive, sender)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/routing.py", line 589, in __call__await route.handle(scope, receive, send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/routing.py", line 250, in handleawait self.app(scope, receive, send)
File "/home/site/wwwroot/.python_packages/lib/site-packages/starlette/routing.py", line 52, in appresponse = await func(request)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/_base/provider.py", line 44, in login_callback
    return self.authenticator.process_login_callback(request)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/_base/authenticators/session.py", line 66, in process_login_callback
    user = self._get_user_from_token(token)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/providers/aad.py", line 109, in _get_user_from_token
    return super()._get_user_from_token(token, options=options)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/_base/authenticators/session.py", line 91, in _get_user_from_token
    validated_claims = self._token_validator.validate_token(token, options=options)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/_base/validators/token.py", line 120, in validate_token
    claims = self._decode_token(token)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/providers/aad.py", line 194, in _decode_token
    jwk_ = self._get_ms_jwk(token)
File "/home/site/wwwroot/.python_packages/lib/site-packages/fastapi_aad_auth/providers/aad.py", line 188, in _get_ms_jwk
    return jwk.loads(key)
File "/home/site/wwwroot/.python_packages/lib/site-packages/authlib/jose/jwk.py", line 6, in loads
    key_set = JsonWebKey.import_key_set(obj)
File "/home/site/wwwroot/.python_packages/lib/site-packages/authlib/jose/rfc7517/jwk.py", line 55, in import_key_set
    raise ValueError('Invalid key set format')
```

**Comment by djpugh** [comment]
Can you try rolling back to an older release and check - wondering if it's the caching introduced in #79 - which version are you using?

**Comment by nickpholden** [comment]
FYI this is caused by latest authlib. Need authljb<1.0.0

**Comment by djpugh** [comment]
Thanks @nickpholden  - I've found the issues and fixed it for ``authlib >= 1.0.0`` as well.
